components/Extensions/src/Tree/FrameworkElementExtensions.LogicalTree.cs

```diff
@@ -1,773 +1,805 @@
+    /// <param name="predicate">The predicatee to use to match the child nodes.</param>
+    /// <returns>The child that was found, or <see langword="null"/>.</returns>
+    public static T? FindChild<T, TState>(this FrameworkElement element, TState state, Func<T, TState, bool> predicate)
+        where T : notnull, FrameworkElement
+    {
+        PredicateByFunc<T, TState> predicateByFunc = new (state, predicate);
+
+        return FindChild<T, PredicateByFunc<T, TState>>(element, ref predicateByFunc);
+    }
+
+    /// <summary>
+    /// Find the first child element matching a given predicate, using a depth-first search.
+    /// </summary>
+    /// <typeparam name="T">The type of elements to match.</typeparam>
+    /// <typeparam name="TPredicate">The type of predicate in use.</typeparam>
+    /// <param name="element">The root element.</param>
+    /// <param name="predicate">The predicatee to use to match the child nodes.</param>
+    /// <returns>The child that was found, or <see langword="null"/>.</returns>
+    private static T? FindChild<T, TPredicate>(this FrameworkElement element, ref TPredicate predicate)
+        where T : notnull, FrameworkElement
+        where TPredicate : struct, IPredicate<T>
+    {
+        // Jump label to manually optimize the tail recursive paths for elements with a single
+        // child by just overwriting the current element and jumping back to the start of the
+        // method. This avoids a recursive call and one stack frame every time.
+        Start:
+
+        if (element is Panel panel)
+        {
+            foreach (UIElement child in panel.Children)
+            {
+                if (child is not FrameworkElement current)
+                {
+                    continue;
+                }
+
+                if (child is T result && predicate.Match(result))
+                {
+                    return result;
+                }
+
+                T? descendant = FindChild<T, TPredicate>(current, ref predicate);
+
+                if (descendant is not null)
+                {
+                    return descendant;
+                }
+            }
+        }
+        else if (element is ItemsControl itemsControl)
+        {
+            foreach (object item in itemsControl.Items)
+            {
+                if (item is not FrameworkElement current)
+                {
+                    continue;
+                }
+
+                if (item is T result && predicate.Match(result))
+                {
+                    return result;
+                }
+
+                T? descendant = FindChild<T, TPredicate>(current, ref predicate);
+
+                if (descendant is not null)
+                {
+                    return descendant;
+                }
+            }
+        }
+        else if (element is ContentControl contentControl)
+        {
+            if (element is Expander expander)

```

Comment:
I'm not sure why Github is picking this up as changing the entire file. I only made some small additions to add cases for the Expander under ContentControl.

Reply:
Is it picking up line ending changes? I'm using VSCode and it should be picking up the editorconfig settings?

Reply:
![image](https://github.com/user-attachments/assets/c0a930bc-c49b-430e-aafb-06cd6a202e1e)